### PR TITLE
rexray: upgrade to v0.11.2 [Backport to 1.11]

### PR DESCRIPTION
**Pull request by gpaul**
_Backport of https://github.com/dcos/dcos/pull/2882 to `1.11`. Contains unit file backport of https://github.com/dcos/dcos/pull/2593/commits/0b66b574f27005c41b5eaeb2b6f2261d6b575113#diff-b4738aec95eb5d18a80de948ec4ce039R16, too.__

## High-level description

What features does this change enable? What bugs does this change fix?

This PR bumps REX-Ray from v0.9.0 to v0.11.2. It configures REX-Ray to operate in `DOCKER_LEGACY` mode as the newer version of REX-Ray switched to CSI-based logic. The REX-Ray developers seem to agree as REX-Ray's master branch was recently switched to `DOCKER_LEGACY` by default: https://github.com/rexray/rexray/commit/6162a4c0a35255414e0fb82e18a8ed44800d5749.

This PR also improves the `delete_ec2_volume.py` cleanup script by adding logging and better error handling.

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS_OSS-3597](https://jira.mesosphere.com/browse/DCOS_OSS-3597) Upgrade REX-Ray to v0.11.2

## Related tickets (optional)

Other tickets related to this change:

  - [DCOS-37441](https://jira.mesosphere.com/browse/DCOS-37441) Triage integration test failures for REX-Ray upgrade
  - [DCOS_OSS-2310](https://jira.mesosphere.com/browse/DCOS_OSS-2310) REX-Ray integration test is flaky after version bump

## Checklist for all PRs

  - [x] Added a comprehensible changelog entry to `CHANGES.md` or explain why this is not a user-facing change:
  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here: This bump was designed to introduce no new test or flakiness. Existing tests should keep working at their current level of flakiness (sad).
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [x] Change log from the last version integrated (this should be a link to commits for easy verification and review): [rexray](https://github.com/rexray/rexray/compare/v0.9.0...v0.11.2)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___

### CODE REVIEW

**Comment by mesosphere-mergebot-dev** [comment]
This repo has @mesosphere-mergebot integration. You can interact with the following commands.
```
@mesosphere-mergebot override-status  
@mesosphere-mergebot merge-it  
@mesosphere-mergebot label [Ready For Review |Holding |Request For Comment |Ship It |Work In Progress] 
@mesosphere-mergebot bump-ee  
```

* PR creators can apply one of  `[Ready For Review |Work In Progress]`. [Owners](https://github.com/dcos/dcos/blob/master/owners.json) can apply any label.

**Comment by gpaul** [comment]
@mesosphere-mergebot bump-ee

**Comment by gpaul** [comment]
@mesosphere-mergebot bump-ee

**Comment by gpaul** [comment]
@mesosphere-mergebot bump-ee

**Comment by mesosphere-mergebot-dev** [comment]
Your pull request's branch is not based on the most recent version of `1.11`. Please rebase your changes against this repo's `1.11` branch.

**Comment by gpaul** [comment]
@mesosphere-mergebot bump-ee

**Comment by gpaul** [comment]
@mesosphere-mergebot bump-ee

**Comment by gpaul** [comment]
@mesosphere-mergebot bump-ee

**Comment by gpaul** [comment]
@mesosphere-mergebot label Ready For Review

File: packages/dcos-integration-test/extra/util/delete_ec2_volume.py [hunk 1]
```diff
@@ -41,29 +45,52 @@ def _remove_env_vars(*env_vars):
 
 @retrying.retry(
     wait_exponential_multiplier=1000,
-    wait_exponential_max=300000,
-    stop_max_delay=1800000,
+    wait_exponential_max=300 * 000,
```
Comment (jdef) [inline packages/dcos-integration-test/extra/util/delete_ec2_volume.py]: typo? `000`

**Comment by gpaul** [comment]
@mesosphere-mergebot bump-ee

**Comment by mesosphere-mergebot-dev** [comment]
Your pull request's branch is not based on the most recent version of `1.11`. Please rebase your changes against this repo's `1.11` branch.

**Comment by gpaul** [comment]
@mesosphere-mergebot bump-ee

**Comment by gpaul** [comment]
@mesosphere-mergebot bump-ee

**Comment by gpaul** [comment]
@mesosphere-mergebot bump-ee

**Comment by gpaul** [comment]
@orsenthil any idea why there aren't any 1.11 trains? What do you think about sending all mergebot logs to a dedicated #mergebot slack channel so we can help debug by ourselves?

**Pull request merged**